R/fa_i.R

```diff
@@ -113,21 +90,52 @@ fa_i <- function(name,
       )
     }
 
-    htmltools::htmlDependencies(icon_tag) <- htmlDependency
+    icon_tag <- htmltools::attachDependencies(icon_tag, html_dependency)
+    icon_tag <- htmltools::browsable(icon_tag)
+
+    return(icon_tag)
+  }
+
+  # Perform verifications on `name` if `verify_fontawesome` is TRUE
+  if (verify_fontawesome) {
+
+    # Determine if the `name` is a Font Awesome v4
+    # icon name and provide a message
+    if (name %in% fa_tbl$v4_name && !(name %in% fa_tbl$name)) {
+
+      # Obtain the version 5 `name` and `full_name`
+      # for messaging purposes
+      v5_name <- fa_tbl[fa_tbl$v4_name == name, ][1, "name"]
+      v5_name_full <- fa_tbl[fa_tbl$v4_name == name, ][1, "full_name"]
+
+      # State that the v4 icon name should be changed to a v5 one
+      message(
+        "The `name` provided ('", name ,"') is deprecated in Font Awesome 5:\n",
+        "* please consider using '", v5_name, "' or '", v5_name_full, "' instead"
+      )
+    }
 
-  } else {
+    # Provide a message if the icon name can't be resolved from
+    # any Font Awesome 4 or 5 names
+    if (!(name %in% fa_tbl$full_name) &&
+        !(name %in% fa_tbl$name) &&
+        !(name %in% fa_tbl$v4_name)
+    ) {
+      message("This Font Awesome icon ('", name, "') does not exist")
```

Comment:
How about adding a message that says to use `verify_fa=FALSE` if the user is providing their own html dependency object in the UI?

Reply:
Great idea. It's now been added.